internal/resources/resources.go

```diff
@@ -97,8 +97,8 @@ func NewContainerResources(id string) *Resources {
 // TODO: make method on Resources struct.
 func ReleaseResources(ctx context.Context, r *Resources, vm *uvm.UtilityVM, all bool) error {
 	if vm != nil {
-		if r.addedNetNSToVM {
-			if err := vm.RemoveNetNS(ctx, r.netNS); err != nil {
+		if r.addedNetNSToVM && vm.NetworkSetup != nil {
```

Comment:
We never nil out vm.NetworkSetup, so wouldn't this always happen then? 

The check that network setup is not nil is also already done in `vm.TeardownNetworking`. We could instead check whether the returned value from `vm.TeardownNetworking` is `ErrNoNetworkSetup`.

Reply:
Yep, just forgot to remove this

Reply:
if we do that, we can unexport `NetworkSetup` from the UVM struct, which I would prefer.

Reply:
Done!